### PR TITLE
[doc] Apply black on the documentation where it makes sense

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Refs: https://github.com/pylint-dev/pylint/pull/8630#discussion_r1181805890

### CODE REVIEW

File: doc/data/messages/c/class-variable-slots-conflict/good.py [hunk 1]
```diff
@@ -1,5 +1,8 @@
 class Person:
-    __slots__ = ("_age", "name",)
+    __slots__ = (
+        "_age",
+        "name",
+    )
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/class-variable-slots-conflict/good.py:5]:
```suggestion
    __slots__ = (
        "_age",
        "name"
    )
```

File: doc/data/messages/i/invalid-all-format/bad.py [hunk 2]
```diff
@@ -1,3 +1,3 @@
-__all__ = ("CONST")  # [invalid-all-format]
+__all__ = "CONST"  # [invalid-all-format]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/invalid-all-format/bad.py:1]: Fortunately this is still invalid.

File: doc/data/messages/i/invalid-slots/good.py [hunk 3]
```diff
@@ -1,2 +1,5 @@
 class Person:
-    __slots__ = ("name", "age",)
+    __slots__ = (
+        "name",
+        "age",
+    )
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/invalid-slots/good.py:5]:
```suggestion
    __slots__ = (
        "name",
        "age"
    )
```

File: doc/data/messages/t/too-many-ancestors/bad.py [hunk 4]
```diff
@@ -1,12 +1,37 @@
-class Animal: ...
-class BeakyAnimal(Animal): ...
-class FurryAnimal(Animal): ...
-class Swimmer(Animal): ...
-class EggLayer(Animal): ...
-class VenomousAnimal(Animal): ...
-class ProtectedSpecie(Animal): ...
-class BeaverTailedAnimal(Animal): ...
-class Vertebrate(Animal): ...
+class Animal:
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-ancestors/bad.py:1]: Might want to disable this one too even if it's not wrong.

File: doc/data/messages/t/too-many-statements/bad.py [hunk 5]
```diff
@@ -1,9 +1,7 @@
 import random
 
 
-def distribute_candies(  # [too-many-statements]
-    children: list[Child], candies_per_child: int
-):
+def distribute_candies(children: list[Child], candies_per_child: int):  # [too-many-statements]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-statements/bad.py:4]: Black is good with the noqa comment !

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -52,6 +52,30 @@ repos:
       - id: black
         args: [--safe, --quiet]
         exclude: *fixtures
+  - repo: https://github.com/psf/black
+    rev: 23.3.0
+    hooks:
+      - id: black
+        name: black-doc
```
Comment (Pierre-Sassoulas) [inline .pre-commit-config.yaml:59]:
```suggestion
        name: black-doc
        # Should be in sync with doc/data/ruff.toml
```

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -52,6 +52,31 @@ repos:
...
+        # Should be in sync with doc/data/ruff.toml
+        args: [--safe, --quiet, -l 103]
+        files: doc/data/messages/
+        exclude: |
```
Comment (DanielNoord) [inline .pre-commit-config.yaml:63]: Can't we re-use this like we do with `&fixtures` and pass it in the `pre-commit` config?

Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml:63]: What need to be in sync is the ``-l 103``, black only looks at ``pyproject.toml`` so we have to give the option like this. (Alternatively we could use the default value, it's not super important.).

I don't think the exclude can be reused, this is the list of the message that black can fix automatically, it's really specific to black.

It's also a very useful list btw ! I'm considering to recover it automatically and generalize it to autofixxer (isort, autoflake, and ruff), in order to use this information directly in our doc for #8579

Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml:63]: I.e. if applying a tool on ``doc/data`` makes the tests fail on a ``m/my-message/bad.py`` file, then the tool autofix ``my-message``. We might have to use it on functional tests instead for complicated fixes like ruff or autoflake to check which use cases is autofixed or not, but for black on the doc is enough and you get the idea.

Comment (DanielNoord) [reply .pre-commit-config.yaml:63]: Yeah, let's use the default then. Less config == better imo

Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml:63]: I'm going to need to rebase and rewrite black from zero, it's not THAT good with comment after all 😄

Comment (DanielNoord) [reply .pre-commit-config.yaml:63]: Shall we merge this and do another PR that removes the config?

Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml:63]: No, it breaks the comments to go back to 88 char per line after doing 103 char per line, it's a lot easier to do it from scratch again :)

File: doc/data/messages/e/exec-used/good.py [hunk 8]
```diff
@@ -1,8 +1,10 @@
...
 
 username = "Ada"
-allowed_globals = {'__builtins__' : None}
-allowed_locals = {'print': print}
-exec(get_user_code(username), allowed_globals, allowed_locals)  # pylint: disable=exec-used
+allowed_globals = {"__builtins__": None}
+allowed_locals = {"print": print}
+exec(  # pylint: disable=exec-used
+    get_user_code(username), allowed_globals, allowed_locals
+)
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/e/exec-used/good.py:10]:
```suggestion
# pylint: disable-next=exec-used
exec(get_user_code(username), allowed_globals, allowed_locals)
```

File: doc/data/messages/r/redefined-argument-from-local/bad.py [hunk 9]
```diff
@@ -1,3 +1,6 @@
 def show(host_id=10.11):
-    for host_id, host in [[12.13, 'Venus'], [14.15, 'Mars']]:  # [redefined-argument-from-local]
+    for host_id, host in [  # [redefined-argument-from-local]
+        [12.13, "Venus"],
+        [14.15, "Mars"],
+    ]:
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/r/redefined-argument-from-local/bad.py:5]:
```suggestion
    # +1: [redefined-argument-from-local]
    for host_id, host in [[12.13, "Venus"], [14.15, "Mars"]]:
```

File: doc/data/messages/s/simplifiable-if-statement/bad.py [hunk 10]
```diff
@@ -2,7 +2,9 @@
 
 
 def is_flying_animal(an_object):
-    if isinstance(an_object, Animal) and an_object in FLYING_THINGS:  # [simplifiable-if-statement]
+    if (  # [simplifiable-if-statement]
+        isinstance(an_object, Animal) and an_object in FLYING_THINGS
+    ):
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/s/simplifiable-if-statement/bad.py:7]:
```suggestion
  # +1: [simplifiable-if-statement]
    if isinstance(an_object, Animal) and an_object in FLYING_THINGS:
```

File: doc/data/messages/s/simplifiable-if-statement/bad.py [hunk 11]
```diff
@@ -2,7 +2,8 @@
 
 
 def is_flying_animal(an_object):
-    if isinstance(an_object, Animal) and an_object in FLYING_THINGS:  # [simplifiable-if-statement]
+  # +1: [simplifiable-if-statement]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/s/simplifiable-if-statement/bad.py:5]:
```suggestion
   # +1: [simplifiable-if-statement]
```

File: doc/data/messages/s/simplifiable-if-statement/bad.py [hunk 12]
```diff
@@ -2,7 +2,8 @@
 
 
 def is_flying_animal(an_object):
-    if isinstance(an_object, Animal) and an_object in FLYING_THINGS:  # [simplifiable-if-statement]
+   # +1: [simplifiable-if-statement]
+    if isinstance(an_object, Animal) and an_object in FLYING_THINGS:
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/s/simplifiable-if-statement/bad.py:6]:
```suggestion
    # +1: [simplifiable-if-statement]
    if isinstance(an_object, Animal) and an_object in FLYING_THINGS:
```

File: .pre-commit-config.yaml [hunk 13]
```diff
@@ -52,6 +52,30 @@ repos:
...
+  - repo: https://github.com/psf/black
+    rev: 23.3.0
+    hooks:
+      - id: black
+        name: black-doc
+        args: [--safe, --quiet]
+        files: doc/data/messages/
+        exclude: |
+          (?x)^(
+              doc/data/messages/r/redundant-u-string-prefix/bad.py|
```
Comment (DanielNoord) [inline .pre-commit-config.yaml:64]: Shall we make this alphabetized?

File: .pre-commit-config.yaml [hunk 14]
```diff
@@ -52,6 +52,30 @@ repos:
       - id: black
         args: [--safe, --quiet]
         exclude: *fixtures
+  - repo: https://github.com/psf/black
```
Comment (DanielNoord) [inline .pre-commit-config.yaml:55]: One final nit, can't this be a separate hook in the `repo` above? Or this is not how `pre-commit` works?

Comment (DanielNoord) [reply .pre-commit-config.yaml:55]: Pushed a small change, let me know what you think!

Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml]: Nice !

Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml]: Same thing could be done for ruff actually

**Review by DanielNoord — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 14 above)
Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml]: I'll do that next time there's a pre-commit autoupgrade, (should be soon).

**Pull request merged**